modules/useRoutes.js

```diff
@@ -244,11 +244,22 @@ function useRoutes(createHistory) {
       })
     }
 
+    function listen(listener) {
+      warning(
```

Comment:
I intentionally use this instead of something like `history/lib/deprecate`; this allows `babel-plugin-dev-expression` to strip out the deprecation message in production builds.